vortex-datafusion/src/persistent/opener.rs

```diff
@@ -78,3 +98,19 @@ impl FileOpener for VortexFileOpener {
         .boxed())
     }
 }
+
+fn apply_plan(
+    file_meta: FileMeta,
+    row_count: u64,
+    scan_builder: ScanBuilder<ArrayRef>,
+) -> ScanBuilder<ArrayRef> {
+    if let Some(byte_range) = file_meta.range.as_ref() {
+        let average_row = file_meta.object_meta.size as u64 / row_count;
```

Comment:
Add a couple of tests just to make sure that these divisions do end up covering all rows from start..end